src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+function prepareItems(
+  allItems: string[],
+  itemsPerPage: number,
+  currentPage: number,
+) {
+  const start = itemsPerPage * currentPage - itemsPerPage;
+  const end =
+    start + itemsPerPage > allItems.length
+      ? allItems.length
+      : start + itemsPerPage;
+
+  return allItems.slice(start, end);
+}
+
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const visibleItems = prepareItems(items, itemsPerPage, currentPage);
+
+  const startItem = itemsPerPage * currentPage - itemsPerPage;
+  const endItem =
+    startItem + itemsPerPage > items.length
+      ? items.length
+      : startItem + itemsPerPage;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startItem + 1} - {endItem} of {items.length})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            onChange={event => {
+              setItemsPerPage(+event.target.value);
+              setCurrentPage(1);
+            }}
+          >
             <option value="3">3</option>
-            <option value="5">5</option>
+            <option value="5" selected>
```

Comment:
Using the `selected` attribute on the `option` tag is not the recommended React way to set a default value for a select element. Instead, you should set the `value` prop on the `select` element to the desired default value, which in this case is the state `itemsPerPage`. This ensures that the select element's value is controlled by React.